### ISSUE TITLE
[boost] <boost>/<1.75.0>: build with python cannot find python include files

### ISSUE THREAD

**Issue by Marcin-Radecki** [conan-io/conan-center-index#4652]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **boost/1.75.0**
  * Operating System+version: **CentOS Linux release 7.8.2003 (Core)**
  * Compiler+version: **GCC 9**
  * Conan version: **conan 1.32.1**
  * Python version: **Python 3.6.8**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
Configuration for profile default:

[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.libcxx=libstdc++11
build_type=RelWithDebInfo
compiler.cppstd=17
os.version=rh7
os_build.release=rh7
compiler.version=9
[options]
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)

Boost recipe cannot find out by itself Python include files, despite fact they are installed on the system (ie, python3 and python3-devel is installed).

`-o python_binary=/bin/python3` must be explicetely set to workaround this. 

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
# Failed run below. Please note that instead of giving python3 binary to command to find python version, it's given the path to conan binary itself, which is incorrect:

conan install boost/1.75.0@ -o without_python=False --build=missing 
....
boost/1.75.0: running "/var/ig/<user>/git/conan-standalone/bin/conan" -c "from __future__ import print_function; import sysconfig; print(sysconfig.get_python_inc())"
boost/1.75.0: (failed)
boost/1.75.0: 
boost/1.75.0: ERROR: Package '7f107f4b7e5623705936db13dd30d13fdedf8eae' build failed
boost/1.75.0: WARN: Build folder /var/ig/<user>/git/conan-standalone/bamboo/.conan/data/boost/1.75.0/_/_/build/7f107f4b7e5623705936db13dd30d13fdedf8eae
ERROR: boost/1.75.0: Error in build() method, line 657
	self._create_user_config_jam(self._boost_build_dir)
while calling '_create_user_config_jam', line 1005
	includes=self._python_includes,
while calling '_python_includes', line 526
	raise Exception("couldn't locate Python.h - make sure you have installed python development files")
	Exception: couldn't locate Python.h - make sure you have installed python development files


# If you give explicit path to python3 binary, then it works:
 
➜  conan-standalone git:(develop) conan install boost/1.75.0@ -o without_python=False -o python_binary=/bin/python3 --build=missing 
...
boost/1.75.0: checking /usr/include/python3.6m/Python.h
boost/1.75.0: found Python.h: /usr/include/python3.6m/Python.h
boost/1.75.0: running "/bin/python3" -c "from __future__ import print_function; import sysconfig; print(sysconfig.get_config_var('LIBRARY'))"
boost/1.75.0: running "/bin/python3" -c "from __future__ import print_function; import sysconfig; print(sysconfig.get_config_var('LDLIBRARY'))"
boost/1.75.0: running "/bin/python3" -c "from __future__ import print_function; import sysconfig; print(sysconfig.get_config_var('LIBDIR'))"
boost/1.75.0: running "/bin/python3" -c "from __future__ import print_function; import sysconfig; print(sysconfig.get_config_var('MULTIARCH'))"
boost/1.75.0: running "/bin/python3" -c "from __future__ import print_function; import sysconfig; print(sysconfig.get_config_var('multiarchsubdir'))"
boost/1.75.0: running "/bin/python3" -c "from __future__ import print_function; import distutils.sysconfig as du_sysconfig; print(du_sysconfig.get_config_var('multiarchsubdir'))"
boost/1.75.0: running "/bin/python3" -c "from __future__ import print_function; import sysconfig; print(sysconfig.get_config_var('WITH_DYLD'))"
boost/1.75.0: running "/bin/python3" -c "from __future__ import print_function; import sys; print('%s.%s' % (sys.version_info[0], sys.version_info[1]))"
boost/1.75.0: running "/bin/python3" -c "from __future__ import print_function; import sys; print(getattr(sys, 'abiflags', ''))"
boost/1.75.0: checking /usr/lib64/libpython3.6m.so
boost/1.75.0: found python library: /usr/lib64/libpython3.6m.so```

</details>

**Comment by sinall** [comment]
Have you addressed the issue? I have very similar problem #8070 .

**Comment by SSE4** [comment]
I hope this problem will be solved after #1510 (should be merged really soon).
it will allow to have a fully controlled and reproducible builds.
right now, specifiying the path to the interpreter via an option is the best choice.

**Comment by perseoGI** [comment]
Hi there, 
I was reviewing old issues and I have come across this one.

I'm closing this issue as this recipe has suffer plenty of modifications since this issue was reported.
If for some reason the problem remains nowadays, do not hesitate to reopen this issue or create a new ticket with updated information, preferably using `conan v2`.

Happy coding 🐸
